drivers/sensor/wsen_tids_2521020222501/wsen_tids_2521020222501.c

```diff
@@ -0,0 +1,330 @@
+			LOG_ERR("Failed to enable one shot");
+			return -EIO;
+		}
+
+		TIDS_state_t busy = TIDS_enable;
+
+		do {
+			if (TIDS_isBusy(&data->sensor_interface, &busy) != WE_SUCCESS) {
+				LOG_ERR("Failed to check for data ready");
+				return -EIO;
+			}
+		} while (TIDS_enable == busy);
+	}
+
+	if (TIDS_getRawTemperature(&data->sensor_interface, &raw_temperature) != WE_SUCCESS) {
+		LOG_ERR("Failed to fetch data sample");
+		return -EIO;
+	}
+
+	data->temperature = raw_temperature;
+
+	return 0;
+}
+
+static int tids_2521020222501_channel_get(const struct device *dev, enum sensor_channel chan,
+					  struct sensor_value *val)
+{
+	struct tids_2521020222501_data *data = dev->data;
+
+	switch (chan) {
+	case SENSOR_CHAN_AMBIENT_TEMP:
+		/* Convert temperature from 0.01 degrees Celsius to degrees Celsius */
+		val->val1 = data->temperature / 100;
+		val->val2 = ((int32_t)data->temperature % 100) * (1000000 / 100);
+		break;
+	default:
+		LOG_ERR("Channel not supported %d", chan);
+		return -ENOTSUP;
+	}
+
+	return 0;
+}
+
+/* Set output data rate. See tids_2521020222501_odr_list for allowed values. */
+static int tids_2521020222501_odr_set(const struct device *dev, const struct sensor_value *odr)
+{
+	struct tids_2521020222501_data *data = dev->data;
+
+	int odr_index;
+
+	for (odr_index = 0; odr_index < ARRAY_SIZE(tids_2521020222501_odr_list); odr_index++) {
+		if (odr->val1 == tids_2521020222501_odr_list[odr_index].val1 &&
+		    odr->val2 == tids_2521020222501_odr_list[odr_index].val2) {
+			break;
+		}
+	}
+
+	if (odr_index == ARRAY_SIZE(tids_2521020222501_odr_list)) {
+		/* ODR not allowed (was not found in tids_2521020222501_odr_list) */
+		LOG_ERR("Bad sampling frequency %d.%d", odr->val1, odr->val2);
+		return -EINVAL;
+	}
+
+	if (odr->val1 == tids_2521020222501_odr_list[0].val1) {
+		if (TIDS_enableBlockDataUpdate(&data->sensor_interface, TIDS_disable) !=
+		    WE_SUCCESS) {
+			LOG_ERR("Failed to enable block data update.");
+			return -EIO;
+		}
+
+		if (TIDS_enableContinuousMode(&data->sensor_interface, TIDS_disable) !=
+		    WE_SUCCESS) {
+			LOG_ERR("Failed to enable continuous mode.");
+			return -EIO;
+		}
+	} else {
+		if (TIDS_setOutputDataRate(&data->sensor_interface,
+					   (TIDS_outputDataRate_t)(odr->val1 - 1)) != WE_SUCCESS) {
+			LOG_ERR("Failed to set output data rate");
+			return -EIO;
+		}
+
+		if (TIDS_enableBlockDataUpdate(&data->sensor_interface, TIDS_enable) !=
+		    WE_SUCCESS) {
+			LOG_ERR("Failed to enable block data update.");
+			return -EIO;
+		}
+
+		if (TIDS_enableContinuousMode(&data->sensor_interface, TIDS_enable) != WE_SUCCESS) {
+			LOG_ERR("Failed to enable continuous mode.");
+			return -EIO;
+		}
+	}
+
+	data->sensor_odr = (uint8_t)odr->val1;
+
+	return 0;
+}
+
+/* Get output data rate. */
+static int tids_2521020222501_odr_get(const struct device *dev, struct sensor_value *odr)
+{
+
+	struct tids_2521020222501_data *data = dev->data;
+
+	TIDS_state_t continuousModeState;
```

Comment:
Please no camel case

Reply:
I made all variables of the form lower case with underscores between different nouns